File: MedicusApp/ClientApp/src/pages/Spec.js
import {useEffect, useState} from "react";
import {Container} from "reactstrap";
import Line from "../components/Line";
import DocsDoctors from "../components/Docs/DocsDoctors";
import Loading from "../components/Loading";
import DocsPricing from "../components/Docs/DocsPricing";
import DocsDescription from "../components/Docs/DocsDescription";

export default function Spec(props) {
  let [data, setData] = useState();

  useEffect(() => {
    fetch("api/Specs/" + props.type)
      .then(r => r.json())
      .then(j => setData(j));
  });

  if(data) {
    return (
      <Container>
        <h2 className="text-center">{data.name}</h2>
        <Line center/>
        <DocsDoctors doctors={data.doctors}/>
        <DocsDescription descriptions={data.descriptions}/>
        <DocsPricing prices={data.prices}/>
      </Container>
    )
  } else {
    return (
      <Loading/>
    )
  }
}

File: MedicusApp/Models/People/Spec.cs
namespace MedicusApp.Models.People
{
    public class Spec
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Href { get; set; }
        public string ClassName { get; set; }
        public int Order { get; set; }
        public List<Doctor> Doctors { get; set; }
        public List<WorkingHours> WorkingHours { get; set; }
        public List<Price> Prices { get; set; }
        public List<Description> Descriptions { get; set; }
    }
}
